src/eureka/S1_detector_processing/ramp_fitting.py

```diff
@@ -87,6 +86,33 @@ def process(self, input):
             Updated for JWST version 1.3.3, code restructure
         '''
         with datamodels.RampModel(input) as input_model:
+
+            if self.s1_meta.grouplevel_bg:
+                from ..S3_data_reduction import background as bkg
+                import astraeus.xarrayIO as xrio
+
+                all_data = input_model.data
+                dq = input_model.groupdq
+                self.s1_meta.inst = input_model.meta.instrument.name.lower()
+                self.s1_meta.int_start = 0
+                self.s1_meta.n_int = all_data.shape[0]
+
+                for ngrp in range(all_data.shape[1]):
+                    grp_data = all_data[:,ngrp,:,:]
+                    grp_mask = np.ones(grp_data.shape, dtype=bool)
+                    dqmask = np.where(dq[:,ngrp,:,:]!= 0)
+                    grp_mask[dqmask] = 0
+
+                    xrdata = (['time', 'y', 'x'], grp_data)
+                    xrmask = (['time', 'y', 'x'], grp_mask)                
+                    xrdict = dict(flux=xrdata, mask=xrmask)
+                    data = xrio.makeDataset(dictionary=xrdict)
+                    data['flux'].attrs['flux_units']='n/a'
+
+                    data = bkg.BGsubtraction(data, self.s1_meta, self.s1_log, self.s1_meta.isplots)
```

Comment:
I assume this line is probably too long for flake8. The following should work fine (and gives you some idea of how flake8 wants things formatted)
```
                    data = bkg.BGsubtraction(data, self.s1_meta, self.s1_log,
                                             self.s1_meta.isplots)
```